### DISCUSSION TITLE
Email notifications editor with variables suggestions

### DISCUSSION THREAD

**Discussion by yingban** [solspace/craft-freeform#1642] [Feature Requests]
Hi,

We had a feature request from a client about the email notification feature:
is there any chance to have an editor/wysiwyg to write the email's body, and suggestions of possible variables?

Thank you!

**Comment by kjmartens** [comment]
Thank you for your feedback @yingban,

The trouble with this is that email notification templates are global to all forms, so this could get complicated as it wouldn't know which form(s) to suggest fields for or would need to list all available fields across all available forms. This might be okay for some setups but could get ugly very fast for larger setups with many unique forms, unfortunately. 🙁 

We'll continue to consider this... there might be a creative solution we can come up with in the future. 🙂

**Reply by yingban** [reply]
Thank you @kjmartens for your answer.

Indeed, it might work (only) if there would be a possibility to add a form based (not global) notification option in the "Notifications" tab, which will be able to use the current form fields. 

The reason of an editor is that we have a lot of non techie users / editors.

**Reply by kjmartens** [reply]
For sure. I understand this use case and we'd like to address it in the future. 🙂

**Comment by kjmartens** [comment]
Just to update this FR, we are now beginning to work on a large update to improve the email notification template experience. An editor will be included as well. 🙂

**Comment by kjmartens** [comment]
This has now been implemented with the new Form-specific email notification templates feature: 🙂 
https://docs.solspace.com/craft/freeform/v5/forms/email-notifications/#form-specific-templates

**Comment by yingban** [comment]
Thanks @kjmartens!
Does it work on Craft 4? I tried searching for it, but could not find it

**Reply by kjmartens** [reply]
It's available in Freeform 5.11 for Craft 4 and 5. 🙂 

Docs are a work in progress, but you can check out:

https://docs.solspace.com/craft/freeform/v5/forms/email-notifications/#template-methods
